Cluster Manager UI & Query Console UI revamp

shahsank3t opened the pull request:
## Description
Complete revamp of existing UI containing Cluster Manager & Query Console.
- This PR is against https://github.com/apache/incubator-pinot/issues/5649
- Cluster Manager UI contains a listing of Tenants & Table within, Broker, Controller, Server, Cluster Configuration
![screencapture-localhost-9000-2020-07-11-02_30_04](https://user-images.githubusercontent.com/6761317/87204309-1d885600-c322-11ea-8d48-0767fd094973.png)
![screencapture-localhost-9000-2020-07-11-02_33_16](https://user-images.githubusercontent.com/6761317/87204317-21b47380-c322-11ea-9cfa-d46847c7635e.png)

- Query Console UI is a complete revamp with Material UI styling.
![screencapture-localhost-9000-2020-07-11-02_35_51](https://user-images.githubusercontent.com/6761317/87204471-84a60a80-c322-11ea-8aa9-1ea61b9e506d.png)

- This PR removes all the existing pinot-controller's static & webapp and gets replaced with an app developed using React + TypeScript + Material UI and few other 3rd Party libraries and built using webpack.

- Also added a plugin in pinot-controller pom file to build the frontend using node & npm.

**Open items that can be added as part of another PR:**
- SQL Hints in SQL Editor
- Reporting error when SQL Query fails

mayankshriv commented:
A couple suggestions:
* For cluster manager, we should be able to drill down at table level as well (apart from tenant).
* It would be super awesome if the UI provides a way to authenticate for certain operations (such as query, write operations on cluster like schema update etc). Note, we already have ACL support in the code, we just need the UI to pass the credentials.

kishoreg commented:
@mayankshriv this is the first cut. lets get this in and we can add more functionality in the subsequent PR

shahsank3t commented:
`testQuery` Test was failing but we longer have static page for query and so removed it.

kishoreg commented:
@shahsank3t query editor does not work well. the cursor keeps jumping/moving to the end as I type.

mayankshriv commented:
> @mayankshriv this is the first cut. lets get this in and we can add more functionality in the subsequent PR

@kishoreg Yes agree, my comment is more towards the general direction of where we should take it. Also, I take it that cluster manager UI will replace Swagger? Would be good to start a doc/issue to capture all the features we would want cluster manager to have and address them in future PRs.

snleee commented:
@shahsank3t  I see that you are deleting a lot of js files. I think that the license file needs to be updated accordingly. Can you fix `LICENSE` file? You will see the following lines:

```

MIT License
-----------
pinot-controller/src/main/resources/*/js/lib/codemirror/*
pinot-controller/src/main/resources/*/js/lib/foundation/*
pinot-controller/src/main/resources/*/js/lib/angular*
pinot-controller/src/main/resources/*/js/lib/underscore*
pinot-controller/src/main/resources/*/js/lib/jquery-2.1.3.min.js
pinot-controller/src/main/resources/*/js/lib/beautify.js
pinot-controller/src/main/resources/*/js/lib/handlebars.js
pinot-controller/src/main/resources/*/css/lib/codemirror*
pinot-controller/src/main/resources/*/css/lib/foundation*
pinot-controller/src/main/resources/*/css/lib/normalize.css

BSD 3-Clause
------------
pinot-controller/src/main/resources/*/js/lib/jquery.dataTables.min.js
```

shahsank3t commented:
@kishoreg 
Good Catch. Fixed the SQL code editor cursor issue.

@snleee 
Updated the existing LICENSE file with the dependencies & dev-dependencies libraries license info. Please have a look and let me know if need to add any more info. Thanks in advance.

shahsank3t commented:
> * For cluster manager, we should be able to drill down at table level as well (apart from the tenant).

@kishoreg @mayankshriv - Addressed this suggestion and added the Tables details page. Yes, currently showing the JSON object and later on can add incremental updates to it.
![screencapture-localhost-8080-2020-07-15-20_18_50](https://user-images.githubusercontent.com/6761317/87560584-3fdef280-c6d9-11ea-9f0f-43a5bf25275a.png)
